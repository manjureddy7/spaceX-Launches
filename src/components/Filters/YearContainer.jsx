import React, { useState } from "react";
import styles from "../../../styles/FilterContainer.module.css";
import Router, { useRouter } from "next/router";
import { LAUNCH_YEAR } from "../constants";

const YearContainer = () => {
  const [activeState, setActiveState] = useState(false);
  const [launchYear, setLaunchYear] = useState(); // This will help us to set active class

  // Get current queries and append to the existing one
  const router = useRouter();

  // Handle Year Filter
  // Based on Active state, update Router
  const handleLaunchYearFilter = (year) => {
    setActiveState(!activeState);
    setLaunchYear(year);
    if (!activeState) {
      Router.push({
        pathname: "/",
        query: { ...router.query, year },
      });
    } else {
      const { year, ...restFilters } = router.query;
      Router.push({
        pathname: "/",
        query: { ...restFilters },
      });
    }
  };
  return (
    <div className={styles.InnerContainer} data-testid="YearContainer">
      <div className={styles.InnerContainerHeader}>Launch Year</div>
      <div className={styles.GroupContainer}>
        {LAUNCH_YEAR.map((year, index) => (
          <div key={index} className={styles.ButtonContainer}>
            <button
              className={
                launchYear === year && activeState ? styles.active : styles.btn
              }
              onClick={() => handleLaunchYearFilter(year)}
            >
              {year}
            </button>
          </div>
        ))}
      </div>
    </div>
  );
};

export default YearContainer;
