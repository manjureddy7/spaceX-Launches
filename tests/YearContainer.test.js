import { render, cleanup } from "@testing-library/react";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import YearContainer from "../components/Filters/YearContainer";
import router from "../helper/helper";

afterEach(cleanup);

describe("YearContainer Component", () => {
  const { getByTestId, asFragment } = render(
    <RouterContext.Provider value={router}>
      <YearContainer />
    </RouterContext.Provider>
  );
  test("Should render correctly on route", async () => {
    expect(getByTestId("YearContainer")).toBeInTheDocument();
  });
  test("Should match snapshot", () => {
    expect(asFragment(<YearContainer />)).toMatchSnapshot();
  });
});
