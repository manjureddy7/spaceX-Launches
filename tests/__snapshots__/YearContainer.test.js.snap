// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearContainer Component Should match snapshot 1`] = `<DocumentFragment />`;
